### PR TITLE
feature: data process crd definition

### PR DESCRIPTION
**Pull request by TrafalgarZZZ**
### Ⅰ. Describe what this PR does
This PR is a preliminary implementation for DataProcess CRD. 

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by xliuqq — APPROVED** [review]

**Review by cheyang — COMMENTED** [review]
/lgtm

File: api/v1alpha1/dataprocess_types.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	Shell *ShellProcessor `json:"shell,omitempty"`
+}
+
+type JobProcessor struct {
+	// PodTemplate defines Pod specification of the DataProcess job.
+	// +optional
+	PodTemplate *corev1.PodTemplate `json:"podTemplate,omitempty"`
+}
+
+type ShellProcessor struct {
```
[the remark this answers is not in the document]
Comment (TrafalgarZZZ) [reply api/v1alpha1/dataprocess_types.go]: Thx for suggestion. Done.

File: api/v1alpha1/dataprocess_types.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	// Script specifies a shell script that will be executed in DataProcess.
+	// +requried
+	Script string `json:"script"`
+
+	// Image specifies the container image where the shell script will be executed.
+	// +required
+	Image string `json:"image"`
```
[the remark this answers is not in the document]
Comment (TrafalgarZZZ) [reply api/v1alpha1/dataprocess_types.go]: Done.

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**